torch_spyre/csrc/spyre_mem.cpp

```diff
@@ -44,6 +48,411 @@
 
 namespace spyre {
 
+struct DMAParameters {
+  const int64_t size_bytes;  // bytes to transfer
+  const off64_t src_offset;
+  const off64_t dst_offset;
+};
+auto get_device_shape(c10::IntArrayRef sizes) -> std::vector<int64_t> {
+  /* Given the CPU Tensor's shape, return the shape of the tensor on Spyre
+   */
+  auto cpu_shape = sizes.vec();
+  std::vector<int64_t> dev_shape;
+  auto stick_size = 64;
+  auto requires_padding = (cpu_shape.back() % stick_size != 0);
+
+  for (int i = 0; i < cpu_shape.size(); i++) {
+    if (i == 0 && cpu_shape.size() == 1) {
+      dev_shape.push_back(stick_size);
+      if (requires_padding) {
+        dev_shape.push_back(1);
+      } else {
+        dev_shape.push_back(cpu_shape.back() / stick_size);
+      }
+    } else if (i == 0) {
+      dev_shape.push_back(stick_size);
+    } else if (i == sizes.size() - 1) {
+      if (cpu_shape.size() <= 2) dev_shape.push_back(cpu_shape[i - 1]);
+      if (requires_padding) {
+        auto padded_shape =
+            cpu_shape.back() + (stick_size - (cpu_shape.back() % stick_size));
+        dev_shape.push_back(padded_shape / stick_size);
+      } else {
+        dev_shape.push_back(cpu_shape.back() / stick_size);
+      }
+      if (cpu_shape.size() > 2) dev_shape.push_back(cpu_shape[i - 1]);
```

Comment:
if I understand this correctly, if the cpu tensor only has 2 dimensions, then you're doing [stick_size, cpu_dim[0], cpu_dim[1]/stick_size], while on dim > 2 (e.g. 3): [stick_size, cpu_dim[0], cpu_dim[2]/stick_size, cpu_dim[1]]?

If this is the case, please add a comment explaining the logic and also add tests somewhere for all the possible cases you're running through here.